Source/Core/Core/Movie.cpp

```diff
@@ -854,9 +854,9 @@ void RecordInput(GCPadStatus* PadStatus, int controllerID)
 
   CheckPadStatus(PadStatus, controllerID);
 
-  EnsureTmpInputSize((size_t)(s_currentByte + 8));
-  memcpy(&(tmpInput[s_currentByte]), &s_padState, 8);
-  s_currentByte += 8;
+  EnsureTmpInputSize((size_t)(s_currentByte + sizeof(ControllerState)));
+  *reinterpret_cast<ControllerState*>(&tmpInput[s_currentByte]) = s_padState;
```

Comment:
Yep, that's correct.

It might seem a little odd at first thought, but consider the following statement about object alignment in section 3.11 ([basic.align] in the C++14 standard).

> Object types have alignment requirements (3.9.1, 3.9.2) which place restrictions on the addresses at which an object of that type may be allocated.

The alignment requirement of a `char` type is guaranteed to be `1`. That said, a char pointer can't be misaligned, allowing bytewise inspection of objects.

Now say there's some arbitrary pointer that has the address `3` and the following is done to it.

```cpp
u16 val = *reinterpret_cast<u16*>(ptr);
```

The alignment requirement of a `u16` is `2`, and therefore can only exist in a well-defined way on addresses that are a multiple of it (i.e. on a correctly aligned address). However, we're saying a `u16` exists at address `3`, making it a misaligned access, which is undefined behavior (on top of breaking the strict aliasing rule).